pkg/logs/tailers/journald/tailer.go

```diff
@@ -307,24 +332,32 @@ func (t *Tailer) shouldDrop(entry *sdjournal.JournalEntry) bool {
 //     ...
 //     }
 //     }
-func (t *Tailer) getContent(entry *sdjournal.JournalEntry) []byte {
-	payload := make(map[string]interface{})
+//
+// This method is modifying the original map from the entry to avoid a copy.
+func (t *Tailer) getContent(entry *sdjournal.JournalEntry) (message.BasicStructuredContent, []byte) {
+	payload := message.BasicStructuredContent{
+		Data: make(map[string]interface{}),
+	}
 	fields := entry.Fields
-	if message, exists := fields[sdjournal.SD_JOURNAL_FIELD_MESSAGE]; exists {
-		payload["message"] = message
-		delete(fields, sdjournal.SD_JOURNAL_FIELD_MESSAGE)
+	var msg string
+	var exists bool
+
+	// remap systemd "MESSAGE" key into the internal "message" representation
+	if msg, exists = fields[sdjournal.SD_JOURNAL_FIELD_MESSAGE]; exists {
+		payload.SetContent([]byte(msg))
+		delete(fields, sdjournal.SD_JOURNAL_FIELD_MESSAGE) // remove it from the root structure
 	}
-	payload["journald"] = fields
+	payload.Data["journald"] = fields
 
-	content, err := json.Marshal(payload)
+	jsonMarshaled, err := json.Marshal(payload.Data)
 	if err != nil {
+		log.Error("can't marshal journald tailed log", err)
 		// ensure the message has some content if the json encoding failed
 		value, _ := entry.Fields[sdjournal.SD_JOURNAL_FIELD_MESSAGE]
-		content = []byte(value)
+		payload.SetContent([]byte(value)) // best effort trying to have something in the log
```

Comment:
I can't think of a good reason JSON marshaling will fail here, but if `v1behavior: false`, this will override the `message` of the strucutred log, which will then attempt to be re-encoded later on. This could create a very odd log struture that I'm not sure is desirable. 

Perhapse we should only invoke this fallback case for `v1behavior: false`?

Reply:
Yeah this err check is really superficial and agreeing with the solution you mention, I'll push this change in next commit.